libs/remote_storage/src/kms.rs

```diff
@@ -0,0 +1,50 @@
+//! A module that provides a KMS implementation that generates and unwraps the keys.
+//!
+
+#![allow(dead_code)]
+
+/// A KMS implementation that does static wrapping and unwrapping of the keys.
+pub struct NaiveKms {
+    account_id: String,
+}
+
+pub struct KeyPair {
+    pub wrapped: Vec<u8>,
+    pub plain: Vec<u8>,
+}
+
+impl NaiveKms {
+    pub fn new(account_id: String) -> Self {
+        Self { account_id }
+    }
+
+    /// Generate a new key in the KMS. Returns the plain and wrapped keys.
+    pub fn generate_key(&self) -> anyhow::Result<KeyPair> {
```

Comment:
I think this should have a different API at this level: encrypt and decrypt.

Also, it should not be in the `remote_storage` crate but in its own crate: `remote_keys` maybe?

I've built [this](https://github.com/neondatabase/neon/compare/arpad/remote_keys) during the offsite, we could have an architecture mirroring that of remote_storage, i.e. `GenericRemoteKey`, etc.

Reply:
moved to a new crate: https://github.com/neondatabase/neon/pull/11599/commits/c450d3224da73b39c1c7b1352fea16388414c2b2